tools/dissector/sample_manager.cpp

```diff
@@ -347,6 +436,51 @@ namespace OpenDDS
         return 0;
       }
     }
+
+    void
+    Sample_Manager::add_protocol_field(
+      int * hf_index,
+      const std::string & full_name, const std::string & short_name,
+      enum ftenum ft, field_display_e fd
+    ) {
+      // Get copies of the names
+      char * _full_name = strdup(full_name.c_str());
```

Comment:
Does strdup belong to std:: or ACE_OS:: or something else?  Also the _leading_underscore looks odd here.

Reply:
Actually it's from [POSIX string.h](http://pubs.opengroup.org/onlinepubs/9699919799/functions/strdup.html)... and I'm pretty sure I knew that when I wrote it, so I don't know why I used it. It allocates and returns a copy of a string, which I need to keep for the rest of execution. I might change it to keep `std::string`s instead and use c_str() but I don't know how much to count on the pointer remaining valid. I can also just use strlen, new and strcpy, which does the same thing as strdup.

I guess the underscore was an attempt to signify it was a copy of the names but yea I can change them to something like `full_name_copy` if I keep the variable.